app/views/notes/_responses.html.erb

```diff
@@ -5,7 +5,10 @@
     <% @responses = DrupalTag.find_nodes_by_type("#{response_type}:#{@node.id}", "note", 200) %>
     <% @unpaginated = true %>
  
-    <p id="<%= response_type %>s"><a class='btn btn-primary btn-lg' href="/post?tags=<%= response_type %>:<%= @node.id %><% if response_type == 'response' %>,hidden:response<% end %><%= ',' + (@tagnames.uniq.delete_if{|x| x.match(":") }).join(',') if @tagnames && @tagnames.length > 0 %>"><%= t('notes._responses.post_' + response_type) %></a></p>
+    <p id="<%= response_type %>s"><a class='btn btn-primary btn-lg' href="/post?tags=<%= response_type %>:<%= @node.id %><% if response_type == 'response' %>,hidden:response<% end %><%= ',' + (@tagnames.uniq.delete_if{|x| x.match(":") }).join(',') if @tagnames && @tagnames.length > 0 %>"><%= t('notes._responses.post_' + response_type) %></a>&nbsp;
+    <% if @node.power_tag('activity') %>
+    	<% if DrupalTag.where(name: @node.power_tag('activity')) %>
+    		<% @responses = DrupalTag.where(name: @node.power_tag('activity')).tag.nodes.first %><% end %><% end %></p>
```

Comment:
OK, - closer, great -- here, I don't think we need an instance variable for `@responses` -- but we could say:

```ruby
<% if @node.power_tag('activity') %>
  <% if DrupalTag.where(name: @node.power_tag('activity')).first.nodes %>
    <%= DrupalTag.where(name: @node.power_tag('activity')).first.nodes.first.path %>
  <% end %>
<% end %>
```

Please also put `end` statements and close `</p>` tags on their own line, with clean indentations (two spaces per indent) -- that makes it a lot easier to read!

See how this is a lot of logic to put into a template? If we imagined we'd use this logic more than just once, we might want to move some of it into the `DrupalTag` model, to simplify the code here. Then we could also unit test the code to ensure it doesn't break in the future. But it's your choice -- if it's just going to be used here, we're OK just doing a little bit of logic in the templates, as long as its readable and efficient.